File: public/docs/_examples/server-communication_toh2/ts/src/app/heroes-service.ts
// #docregion
import {Inject} from 'angular2/angular2';
import {Http} from 'angular2/http';

export class HeroesService {
  //FIXME: Figure out why it doesn't work with @Injectable()
  constructor (@Inject(Http) private http: Http) {}

    getHeroes () {
    return this.http.get('/heroes').map(res => res.json().items);
  }

  // #docregion addhero
  addHero (name: string) {
    return this.http.post('/heroes', JSON.stringify({ name: 'name' }));
  }
  // #enddocregion addhero
}
// #enddocregion

File: public/docs/_examples/hierarchical-dependency-injection/dart/lib/heroes_service.dart
// #docregion
import 'package:angular2/core.dart';

import 'hero.dart';

@Injectable()
class HeroesService {
  List<Hero> _heroes = [
    new Hero()
      ..name = "RubberMan"
      ..power = 'Flexibility',
    new Hero()
      ..name = "Tornado"
      ..power = 'Weather changer'
  ];

  List<Hero> getHeroes() {
    return _heroes;
  }
}
